packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -571,16 +571,27 @@ export function listOperationsInOperationGroup(
   return operations;
 }
 
+interface CreateSdkContextOptions {
+  ignoreVersioning?: boolean;
+}
+
 export function createSdkContext<
   TOptions extends Record<string, any> = SdkEmitterOptions,
   TServiceOperation extends SdkServiceOperation = SdkHttpOperation,
->(context: EmitContext<TOptions>, emitterName?: string): SdkContext<TOptions, TServiceOperation> {
+>(
+  context: EmitContext<TOptions>,
+  emitterName?: string,
+  options?: CreateSdkContextOptions
+): SdkContext<TOptions, TServiceOperation> {
   const diagnostics = createDiagnosticCollector();
   const protocolOptions = true; // context.program.getLibraryOptions("generate-protocol-methods");
   const convenienceOptions = true; // context.program.getLibraryOptions("generate-convenience-methods");
   const generateProtocolMethods = context.options["generate-protocol-methods"] ?? protocolOptions;
   const generateConvenienceMethods =
     context.options["generate-convenience-methods"] ?? convenienceOptions;
+  if (options?.ignoreVersioning === false) {
+    context.options["api-version"] ?? "all";
```

Comment:
that's not going to set the value here

Reply:
where is it used? also does `all` makes any sense for the others?

Reply:
`all` means we don't do any version projection, we just return all versions. Thanks for pointing out that i'm not actually setting `api-version`. I think it's fair to just set `api-version` here, so we don't need multiple boolean statements down the pipeline

Reply:
juste note that setting this exact object is a bad pattern you are updating an object passed by the consumer that they wouldn't expect to be modified. This could lead to some bug down the line, unlikely but it could.